MdePkg/Library/CompilerIntrinsicsLib/AArch64/Atomics.S

```diff
@@ -32,12 +32,26 @@
         .macro          fn_start, name:req
         .section        .text.\name
         .globl          \name
+
+        /*
+         * PE targets (e.g. building with CLANGPDB) do not support ELF style
+         * .type directives
+         */
+#ifdef __ELF__
         .type           \name, %function
+#endif

```

Comment:
Also use `_ASM_TYPE` here, or if that doesn't work inside the `.macro` for any reason, conditionally define another GAS macro and call that.